packages/scale-codec/src/codecs/boolean.ts

```diff
@@ -0,0 +1,11 @@
+import { Codec } from "../types"
+import { u8 } from "./u8"
+import { enhanceCodec } from "../utils"
+
+const booleanToNumber = (value: boolean) => (value ? 1 : 0)
+
+export const boolean: Codec<boolean> = enhanceCodec(
```

Comment:
I'm not sure if I completely understand this comment... But I _think_ that's already possible? It could have been written like this:

```ts
export const boolean = enhanceCodec<number, boolean>(
  u8,
  booleanToNumber,
  Boolean,
)
```

The thing is that `enhanceCoded` is just a helper function for creating a Codec from an already existing one. In this case, the "base" codec is `u8` which is a `Codec<number>`, so `enhanceCodec` infers the type of the "base" codec and it also infers the type of the resulting codec, also its typings ensure that the `fromTo` and the `toFrom` functions take the correc input and return the correct output.

If I wanted to adjust `enhanceCodec` so that it only accepted the desired output type (without also passing the input type), then the function wouldn't be able to infer the type of the "base" codec and that would mean that the function wouldn't be that helpful because it couldn't ensure that the `fromTo` and the `toFrom` functions are correct. Meaning that the current typings of `enhanceCodec` are:

```ts
const enhanceCodec: <I, O>(
  [encoder, decoder]: Codec<I>,
  toFrom: (value: O) => I,
  fromTo: (value: I) => O
) => Codec<O>
```
AFAIK, changing it so that it would only require the output type would be dangerous, because then we would have to do this, right?:

```ts
const enhanceCodec: <O, I = any>(
  [encoder, decoder]: Codec<I>,
  toFrom: (value: O) => I,
  fromTo: (value: I) => O
) => Codec<O>
```
But now, the `toFrom` and `fromTo` functions could be incompatible and TS wouldn't warn me, right?

Reply:
Also, adding the `: Codec<boolean>` on the definition of `const boolean` is kinda redundant because that's the already inferred type, however, I put it there because I rather being explicit on public exports...

Reply:
Makes complete sense!